### ISSUE TITLE
Metrics: query for entrypoint=none and referrer

### ISSUE THREAD

**Issue by vladikoff** [mozilla/fxa-content-server#6756]
We have this https://github.com/mozilla/fxa-content-server/blob/master/app/scripts/lib/metrics.js#L143

We want to track down `entrypoint=none` integrations and maybe `utm_source=none`

@irrationalagent is there a way to redash this into a table of referres that send `none`?

@davismtl do we want to minimize `entrypoint=none` or `utm_source=none`? For example should we ask all reliers to provide utm_source?

cc @shane-tomlinson

**Comment by davismtl** [comment]
So, we've sort of failed at distinguishing entrypoint vs utm_source in the past. We have an opportunity to really improve ourselves here.

I think there should ALWAYS be an entrypoint. vs. There should normally be a utm_source. What's the difference?

An entrypoint is the touchpoint of FxA that a user entered through. In other words, where in the product are they engaging with us? There will always be an entrypoint! 

The utm_source should be how they got to that entrypoint. 

A hypothetical example could be a Mozilla.org web page that promotes FxA/Sync and drives logins and registrations. The entrypoint should be the name of this page. The utm_source (and other utm params) should describe how users got to that page. Did they arrive from AMO? Did they arrive from within the Firefox product? Did they arrive through a campaign or newsletter? These are what the utm parameters should be used for. This is why it is possible that we don't always have them. For web, where there is no utm params, it's usually because it's organic and direct traffic.

I know that we don't do this very well right now but this is really where I think we should be heading in terms of standardizing our parameters.

**Comment by shane-tomlinson** [comment]
> I know that we don't do this very well right now but this is really where I think we should be heading in terms of standardizing our parameters.

I strongly agree with what you wrote here. Obviously our relier docs should reflect this position, and maybe we can go further by using the dev console (or even something more explicit) to remind oauth relier developers they should specify an entrypoint.

**Comment by irrationalagent** [comment]
I just posted this in IRC, something I realized, maybe I need to file a separate issue.

I think I've seen some cases where entrypoint is defined (for example) in the first few steps of a funnel (view events) but not for later steps. This causes entrypoint to get set as a user property which persists, allowing us to track the origin of a user's registration. I wonder then, if we eliminate all null values by filling in with service/client_id, then will subsequent funnel steps now clobber the user property undesirably?

see [this for example](https://analytics.amplitude.com/mozilla-corp/chart/new/kp0lr36 ) the about:welcome (`activity-stream-firstrun`) entrypoint is defined for the email-first view but not subsequent steps. If a user completes registration through that entrypoint, then they still receive the correct entrypoint as a user property, since all subsequent steps did not have it set. 

If we start filling in the null values for the post-email-first steps with `sync` or something else, I think they would then clobber `activity-stream-firstrun` ?

this seems to be especially true for self-hosted email forms, which are now becoming the norm.
